docs/fsharp/language-reference/results.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Results (F#)
+description: Results (F#)
+keywords: visual f#, f#, functional programming
+author: cartermp
+ms.author: phcart
+ms.date: 04/24/2017
+ms.topic: language-reference
+ms.prod: .net
+ms.technology: devlang-fsharp
+ms.devlang: fsharp
+ms.assetid: a15b5cf1-9055-4481-918c-4c8a051b5829
+---
+
+# Results
+
+Starting with F# 4.1, there is a `Result<'T,'TFailure>` type which you can use for writing error-tolerant code which can be composed monadically.
+
+## Syntax
+
+```fsharp
+// The definition of Result in FSharp.Core
+[<StructuralEquality; StructuralComparison>]
+[<CompiledName("FSharpResult`2")>]
+[<Struct>]
+type Result<'T,'TError> = 
+    | Ok of ResultValue:'T 
+    | Error of ErrorValue:'TError
+```
+
+## Remarks
+
+Note that the result type is a [struct discriminated union](discriminated-unions.md#struct-discriminated-unions), which is another feature introduced in F# 4.1.  Structural equality semantics apply here.
+
+The `Result` type is typically used in monadic error-handling, which is often referred to as [Railway-oriented Programming](https://swlaschin.gitbooks.io/fsharpforfunandprofit/content/posts/recipe-part2.html) within the F# community.  The following trivial example demonstrates this approach.
+
+```fsharp
+// Define a simple type which has fields that can be validated
+type Request = {
+    Name: string
+    Email: string
+}
+
+// Define an infix operator for Result.bind to compose
+let (<<=) input f = bind f input
```

Comment:
In your suggestion, I think it should be:

```fsharp
let validateRequest req =
    req 
    |> validateName
    |> Result.bind validateEmail 
```

Since `req` is a `Request` type and not a `Ok` or `Error` case.